pyvrp/cpp/Individual.cpp

```diff
@@ -238,13 +227,22 @@ Load Individual::Route::demand() const { return demand_; }
 
 Load Individual::Route::excessLoad() const { return excessLoad_; }
 
-Duration Individual::Route::duration() const { return duration_; }
+Duration Individual::Route::travelDuration() const { return travel_; }
 
 Duration Individual::Route::serviceDuration() const { return service_; }
 
-Duration Individual::Route::timeWarp() const { return timeWarp_; }
+Duration Individual::Route::timeWarp() const { return tws_.timeWarp(); }
+
+Duration Individual::Route::waitDuration() const
+{
+    return tws_.duration() - travel_ - service_;
+}
+
+Duration Individual::Route::totalDuration() const { return tws_.duration(); }
+
+Duration Individual::Route::earliestStart() const { return tws_.twEarly(); }
 
-Duration Individual::Route::waitDuration() const { return wait_; }
+Duration Individual::Route::latestStart() const { return tws_.twLate(); }
```

Comment:
Can we support something like this using a `slack()` attribute that's the difference between `twEarly` and `twLate`? Then `start()` can unambiguously be the earliest start time.

Reply:
I think it's useful to add an explicit `slack()` attribute, however I'm not sure if we should define a default start time. Any start time between `earliestStart()` and `latestStart()` is a valid start time that minimizes the route duration, and having an attribute `start()` equal to the earliest possible start time is a (somewhat) arbitrary default and makes it easy for a user to overlook the slack. Also, client time windows are modeled as `earliest` and `latest` starts, not as `start` and `slack`, so I I prefer to model start times similarly and keep it like this, forcing the user to pick an explicit start time from the interval. What do you think?

Reply:
@leonlan what do you think?

Reply:
I think it's easier for users if we have a `start()` time and refer to `slack()` in the docstring, mentioning that the routes can start later.

Reply:
Ok, just to confirm you want `start()` and `slack()` but not `earliest`/`latest`, but keep client time windows as `twEarly` + `twLate` (rather than e.g. `twStart` + `twSlack`)? That means we implicitly pick/suggest the earliest start time as a default if a user ignores slack?

Reply:
Yes.